src/Compilers/Core/Portable/MetadataReference/PortableExecutableReference.cs

```diff
@@ -146,18 +146,30 @@ internal sealed override MetadataReference WithPropertiesImplReturningMetadataRe
         /// 
         /// The calling <see cref="Compilation"/> doesn't take ownership of the <see cref="Metadata"/> objects returned by this method.
         /// The implementation needs to retrieve the object from a provider that manages their lifetime (such as metadata cache).
-        /// The <see cref="Metadata"/> object is kept alive by the <see cref="Compilation"/> that called <see cref="GetMetadata"/>
+        /// The <see cref="Metadata"/> object is kept alive by the <see cref="Compilation"/> that called <see cref="GetMetadataNoCopy"/>
         /// and by all compilations created from it via calls to With- factory methods on <see cref="Compilation"/>, 
         /// other than <see cref="Compilation.WithReferences(MetadataReference[])"/> overloads. A compilation created using 
-        /// <see cref="Compilation.WithReferences(MetadataReference[])"/> will call to <see cref="GetMetadata"/> again.
+        /// <see cref="Compilation.WithReferences(MetadataReference[])"/> will call to <see cref="GetMetadataNoCopy"/> again.
         /// </remarks>
         protected abstract Metadata GetMetadataImpl();
 
-        internal Metadata GetMetadata()
+        internal Metadata GetMetadataNoCopy()
         {
             return GetMetadataImpl();
         }
 
+        /// <summary>
+        /// Returns a fresh copy of the <see cref="Metadata"/> referenced by this <see cref="MetadataReference"/>.
+        /// The instance will not be be the exact same one used by the compiler.  As such, disposing of the
+        /// metadata returned by this will have no effect on the metadata used by existing compilation instances.
+        /// </summary>
+        /// <exception cref="IOException"/>
+        /// <exception cref="BadImageFormatException"/>
+        public Metadata GetMetadata()
```

Comment:
Also, it looks like someone else can still dispose underlying data for the copy. That should probably be reflected in the comment. Basically you own the returned object, but not the underlying data.